code/modules/research/server.dm

```diff
@@ -187,3 +196,116 @@
 	playsound(src, "sparks", 75, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
 	obj_flags |= EMAGGED
 	to_chat(user, span_notice("You disable the security protocols."))
+
+/// Master R&D server. As long as this still exists and still holds the HDD for the theft objective, research points generate at normal speed. Destroy it or an antag steals the HDD? Half research speed.
+/obj/machinery/rnd/server/master
+	var/obj/item/computer_hardware/hard_drive/cluster/hdd_theft/source_code_hdd
+	var/deconstruction_state = HDD_PANEL_CLOSED
+	var/front_panel_screws = 4
+	var/hdd_wires = 6
+
+/obj/machinery/rnd/server/master/Initialize()
+	. = ..()
+	name = "\improper Master " + name
+	source_code_hdd = new(src)
+	SSresearch.master_servers += src
+
+	add_overlay("RD-server-objective-stripes")
+
+/obj/machinery/rnd/server/master/Destroy()
+	if(source_code_hdd)
+		QDEL_NULL(source_code_hdd)
+
+	SSresearch.master_servers -= src
+
+	return ..()
+
+/obj/machinery/rnd/server/master/attackby(obj/item/attacking_item, mob/user, params)
```

Comment:
Check combat mode here as well, otherwise I think it's impossible to attack it normally.

Reply:
Changed to use tool_acts. See commit history for updates.